src/mapper/prelude.rs

```diff
@@ -98,27 +29,31 @@ pub struct Config {
 pub(crate) struct EventWriter {
     context: EventContext,
     output: StageSender,
-    time_provider: Option<NaiveTime>,
-    pub(crate) bech32_provider: Bech32Provider,
     pub(crate) config: Config,
+    pub(crate) utils: Arc<Utils>,
 }
 
 impl EventWriter {
-    pub fn new(
-        output: StageSender,
-        well_known: Option<ChainWellKnownInfo>,
-        config: Config,
-    ) -> Self {
+    pub fn new(output: StageSender, utils: Arc<Utils>, config: Config) -> Self {
         EventWriter {
             context: EventContext::default(),
             output,
-            time_provider: well_known.clone().map(|x| NaiveTime::new(x.into())),
-            bech32_provider: well_known
-                .map_or_else(Bech32Provider::default, |x| Bech32Provider::new(x.into())),
+            utils,
             config,
         }
     }
 
+    #[allow(unused)]
+    pub fn standalone(
+        output: StageSender,
+        well_known: Option<ChainWellKnownInfo>,
+        config: Config,
+    ) -> Self {
+        let utils = Arc::new(Utils::new(well_known.unwrap_or_default()));
+
+        Self::new(output, utils, config)
+    }
+
```

Comment:
This works for me. I renamed my scratch branch to avoid any future confusion. It's not supposed to get merged ever, just to provide my context to other developers. I will delete it once it's consumed by actual change requests to either oura or cardano-ouroboros-network.

https://github.com/txpipe/oura/tree/scratch/pavlix/networklib